### ISSUE TITLE
Properly document exported functions

### ISSUE THREAD

**Issue by JohnStrunk** [gluster/gluster-csi-driver#55]
**Describe the bug**
Some golang exported functions are not properly documented, causing golint to fail.

**Steps to reproduce**

1. run `gometalinter -j4 --sort=path --sort=line --sort=column --deadline 1h --vendor --disable-all --enable golint ./...`

**Actual results**
golint returns errors:
```
pkg/glusterfs/controllerserver.go:27:6:warning: exported type ControllerServer should have comment or be unexported (golint)
pkg/glusterfs/controllerserver.go:377:1:warning: comment on exported method ControllerServer.CreateSnapshot should be of the form "CreateSnapshot ..." (golint)
pkg/glusterfs/controllerserver.go:382:1:warning: comment on exported method ControllerServer.DeleteSnapshot should be of the form "DeleteSnapshot ..." (golint)
pkg/glusterfs/controllerserver.go:387:1:warning: comment on exported method ControllerServer.ListSnapshots should be of the form "ListSnapshots ..." (golint)
pkg/glusterfs/driver.go:16:1:warning: comment on exported type GfDriver should be of the form "GfDriver ..." (with optional leading article) (golint)
pkg/glusterfs/driver.go:45:1:warning: exported function NewControllerServer should have comment or be unexported (golint)
pkg/glusterfs/driver.go:51:1:warning: exported function NewNodeServer should have comment or be unexported (golint)
pkg/glusterfs/driver.go:57:1:warning: exported function NewidentityServer should have comment or be unexported (golint)
pkg/glusterfs/driver.go:63:1:warning: exported method GfDriver.Run should have comment or be unexported (golint)
pkg/glusterfs/identityserver.go:10:6:warning: exported type IdentityServer should have comment or be unexported (golint)
pkg/glusterfs/nodeserver.go:17:6:warning: exported type NodeServer should have comment or be unexported (golint)
pkg/glusterfs/nodeserver.go:131:1:warning: exported method NodeServer.NodeGetId should have comment or be unexported (golint)
pkg/glusterfs/nodeserver.go:131:23:warning: method NodeGetId should be NodeGetID (golint)
```

**Expected behavior**
golint should succeed, and all exported symbols should be properly documented or unexported. No exceptions.

**Additional context**
As a workaround, in PR #53 , I have added stub comments to permit the CI job to pass.
Depends on #53

**Comment by humblec** [comment]
Sure @JohnStrunk , on it!

**Comment by JohnStrunk** [comment]
This was fixed by #58
